src/libc-wrappers/libc-wrappers.c

```diff
@@ -15,7 +15,153 @@
  */
 
 
+#include <pthread.h>
 #include <signal.h>
+#include <link.h>
+
+
+/*
+ * __libc_start_main
+ */
+
+
+struct startup_info
+{
+  void *sda_base;
+  int (*main) (int, char **, char **, void *);
+  int (*init) (int, char **, char **, void *);
+  void (*fini) (void);
+};
+
+extern int __libc_start_main (int argc,
+                              char **argv,
+                              char **ev,
+                              ElfW (auxv_t) * auxvec,
+                              void (*rtld_fini) (void),
+                              struct startup_info * stinfo,
+                              char **stack_on_entry);
+
+
+#if defined __aarch64__
+__asm__(".symver __libc_start_main,__libc_start_main@GLIBC_2.17");
+#elif defined __arm__
+__asm__(".symver __libc_start_main,__libc_start_main@GLIBC_2.4");
+#elif defined __i386__
+__asm__(".symver __libc_start_main,__libc_start_main@GLIBC_2.0");
+#elif defined __powerpc64__ && _CALL_ELF == 2 /* ppc64le */
+__asm__(".symver __libc_start_main,__libc_start_main@GLIBC_2.17");
+#elif defined __s390x__
+__asm__(".symver __libc_start_main,__libc_start_main@GLIBC_2.2");
+#elif defined __x86_64__
+__asm__(".symver __libc_start_main,__libc_start_main@GLIBC_2.2.5");
+#else
+#error "Please specify symbol version for __libc_start_main"
+#endif
+
+
+int
+__wrap___libc_start_main (int argc,
+                          char **argv,
+                          char **ev,
+                          ElfW (auxv_t) * auxvec,
+                          void (*rtld_fini) (void),
+                          struct startup_info * stinfo,
+                          char **stack_on_entry)
+{
+  return __libc_start_main (argc, argv, ev, auxvec, rtld_fini, stinfo, stack_on_entry);
+}
+
+
+/*
+ * pthread_attr_getstacksize
```

Comment:
While going through the glibc Git history, I realized that it would be useful to note the version of glibc that made us add the wrappers.

Currently, the names of the functions serve as a nice touch to divide the file into sections, but the names are still relatively easy to figure out from reading the code. However, the actual glibc version that we are trying to avoid is not clear at all. In the future, when our baseline glibc requirement gets updated, we would be able to drop some of the wrappers, and having the versions noted in the comment will make it easier.

Reply:
I took the liberty to update the branch.